packages/async-flow/src/types.d.ts

```diff
@@ -0,0 +1,252 @@
+import type { Passable } from '@endo/pass-style';
+import type { Vow } from '@agoric/vow';
+import type { LogStore } from './log-store.js';
+import type { Bijection } from './bijection.js';
+
+export type FlowState =
+  | 'Running'
+  | 'Sleeping'
+  | 'Replaying'
+  | 'Failed'
+  | 'Done';
+
+/**
+ * `T` defaults to `any`, not `Passable`, because unwrapped guests include
+ * non-passables, like unwrapped functions and unwrapped state records.
+ * (Unwrapped functions could be made into Remotables,
+ * but since they still could not be made durable, in this context
+ * it'd be pointless.)
+ */
+export type Guest<T extends unknown = any> = T;
+export type Host<T extends Passable = Passable> = T;
+
+/**
+ * A HostVow must be durably storable. It corresponds to an
+ * ephemeral guest promise.
+ */
+export type HostVow<T extends Passable = Passable> = Host<Vow<T>>;
+
+export type GuestAsyncFunc = (
+  ...activationArgs: Guest[]
+) => Guest<Promise<any>>;
+
+export type HostAsyncFuncWrapper = (...activationArgs: Host[]) => HostVow;
+
+/**
+ * The function from the host as it will be available in the guest.
+ *
+ * Specifically, Vow return values are converted to Promises.
+ */
+export type GuestOf<F extends HostAsyncFuncWrapper> = F extends (
+  ...args: infer A
+) => Vow<infer R>
+  ? (...args: A) => Promise<R>
+  : F;
+
+/**
+ * Convert an entire Guest interface into what the host will implement.
+ */
+type HostInterface<T> = {
+  [K in keyof T]: HostOf<T[K]>;
+};
+
+/**
+ * The function the host must provide to match an interface the guest expects.
+ *
+ * Specifically, Promise return values are converted to Vows.
+ */
+export type HostOf<F> = F extends (...args: infer A) => Promise<infer R>
+  ? (...args: A) => Vow<R extends Passable ? R : HostInterface<R>>
+  : F;
+
+export type PreparationOptions = {
+  vowTools?:
+    | {
+        when: <
+          T,
+          TResult1 = import('@agoric/vow').EUnwrap<T>,
+          TResult2 = never,
+        >(
+          specimenP: T,
+          onFulfilled?:
+            | ((
+                value: import('@agoric/vow').EUnwrap<T>,
+              ) => TResult1 | PromiseLike<TResult1>)
+            | undefined,
+          onRejected?:
+            | ((reason: any) => TResult2 | PromiseLike<TResult2>)
+            | undefined,
+        ) => Promise<TResult1 | TResult2>;
```

Comment:
Shouldn't all the vowTools types already be provided by the @agoric/vow package?

Reply:
yep, I'll revise by importing that. this was a mechanical transformation.